go/kbfs/data/path_part_string.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package data
+
+import (
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const (
+	prefixToSkipObfsucation = ".kbfs_"
+	tarGzSuffix             = ".tar.gz"
+
+	// extRestoreMaxLen specifies the max length of an extension, such
+	// that it's restored to the obfuscated string.  It includes the
+	// leading dot.
+	extRestoreMaxLen = 5
+)
+
+// PathPartString returns an obfuscated version of part of a path,
+// preserving the suffixes if they're small enough or if it's a
+// conflicted copy of a file.  Two `PathPartString`s for the asme
+// plaintext data, created using the same `Obfuscator`, should pass
+// equality checks.
+type PathPartString struct {
+	plaintext  string
+	ext        string
+	obsfucator Obfuscator
+}
+
+var _ fmt.Stringer = PathPartString{}
+
+// NewPathPartString creates a new `PathPartString` instance, given a
+// plaintext string representing one part of a path, and a
+// `Obfuscator`.
+func NewPathPartString(
+	plaintext string, obfuscator Obfuscator) PathPartString {
+	_, ext := SplitFileExtension(plaintext)
+	if len(ext) > extRestoreMaxLen && ext != tarGzSuffix {
```

Comment:
why make an exception for `.tar.gz` [and not for `.xcodeproj` / `.tar.bz2` / all the other things that are more than 5 characters]? Just seems sorta random.

Reply:
Good question, I was just following the logic that @taruti wrote in the original extension parsing for conflict resolution.  I guess it was just an arbitrary compromise at the time.  That code (now in `SplitFileExtension`) would not return `tar.bz2` (just `.bz2`).  It would return `.xcodeproj`, but here I'm just trying to be sensitive to people who just put dots in files names rather than underscores, where the last part of the name might be meaningful.  I'm just guessing that for debugging purposes, only the usual short extensions will be useful to us, and so it's not worth the privacy risk for catching longer ones that aren't specially flagged by existing code already.

I can add a comment to that effect.